### PR TITLE
[Test Coverage] Smtp library 

### LINKED ISSUE #1606: [Test Coverage] Add unit tests for Smtp library
The Smtp library currently lacks dedicated unit tests. Add a test file validates successful execution and expected return values.

### LINKED ISSUE #1607: [Test Coverage] Add unit tests for Smtps library
The Smtps library currently lacks dedicated unit tests. Add a test file validates successful execution and expected return values.

### PR DESCRIPTION
**Pull request by Aarush289**
## Proposed change

This PR adds unit tests for Smtp library .
Fixes #1606 

## Type of change

- [ ] New core framework functionality
- [ ] Bugfix (non-breaking change that fixes an issue)
- [ ] Code refactoring without any functionality changes
- [ ] New or existing module/payload change
- [ ] Documentation/localization improvement
- [x] Test coverage improvement
- [ ] Dependency upgrade
- [ ] Other improvement (best practice, cleanup, optimization, etc)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've **digitally signed** all my commits in this PR
- [x] I've run `make pre-commit` and confirm it didn't generate any warnings/changes
- [x] I've run `make test` and I confirm all tests passed locally
- [x] I've added/updated any relevant documentation in the `docs/` folder 
- [x] I've linked this PR with an open issue
- [x] I've tested and verified that my code works as intended and resolves the issue as described
- [ ] I've attached screenshots demonstrating that my code works as intended (if applicable)
- [ ] I've checked all other open PRs to avoid submitting duplicate work
- [x] I confirm that the code and comments in this PR are not direct unreviewed outputs of AI
- [x] I confirm that I am the Sole Responsible Author for every line of code, comment, and design decision

[contributing-guidelines]: https://nettacker.readthedocs.io/en/latest/Developers/#contribution-guidelines

### CODE REVIEW

File: tests/core/lib/test_smtp.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            ),
+            {
+                "host": HOST,
+                "port": PORT,
+                "username": USERNAME,
+                "password": PASSWORD,
+            },
+        )
+
+        mock_smtp_client.assert_called_once_with(HOST, PORT, timeout=TIMEOUT)
```
[the remark this answers is not in the document]
Comment (Aarush289) [reply tests/core/lib/test_smtp.py]: done

File: tests/core/lib/test_smtp.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+class TestSmtpMethod(TestCase):
+    @patch("nettacker.core.lib.smtp.SmtpLibrary.client")
+    def test_brute_force_password(self, mock_smtp_client):
+        library = SmtpLibrary()
+        HOST = "mail.example.com"
+        PORT = SMTP_SESSION_PORT
+        USERNAME = "admin"
+        PASSWORD = "Password@123"
+        TIMEOUT = 5
```
[the remark this answers is not in the document]
Comment (Aarush289) [reply tests/core/lib/test_smtp.py]: done

File: tests/core/lib/test_smtp.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class TestSmtpMethod(TestCase):
+    @patch("nettacker.core.lib.smtp.SmtpLibrary.client")
+    def test_brute_force_password(self, mock_smtp_client):
+        library = SmtpLibrary()
+        HOST = "mail.example.com"
+        PORT = SMTP_SESSION_PORT
+        USERNAME = "admin"
+        PASSWORD = "Password@123"
```
[the remark this answers is not in the document]
Comment (Aarush289) [reply tests/core/lib/test_smtp.py]: done

**Review by NSK-394 — APPROVED** [review]

**Review by securestep9 — APPROVED** [review]

**Comment by securestep9** [comment]
Thank you for your contribution @Aarush289 🚀  ! Note that more tests will be needed in the future to test the negative paths as well
